main.cpp

```diff
@@ -115,19 +122,22 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T t1(10 , "a");                                             //6
+    T t2(20 , "b");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    struct1 f;                                            //7
+    auto* smaller = f.compare(&t1 , &t2);                              //8
+    std::cout << "the smaller one is << " << smaller->name << std::endl; //9 
```

Comment:
What do we always check before using pointers? 
- [ ] revise

Reply:
Always check to make sure your pointer is not NULL before using it!!!! :)